### DISCUSSION TITLE
make nitro configurable

### DISCUSSION THREAD

**Discussion by icai** [nuxt/nuxt#23203] [Questions] [answered]
### Describe the feature

<img width="417" alt="image" src="https://github.com/nuxt/nuxt/assets/1061012/bd94221a-60b0-4ad4-be7b-fa9509787d4b">

I found this in the documentation, "Server [API endpoints](https://nuxt.com/docs/guide/directory-structure/server#api-routes) and [Middleware](https://nuxt.com/docs/guide/directory-structure/server#server-middleware) are added by Nitro that internally uses [h3](https://github.com/unjs/h3). "

but my `Middleware` directory isn't the Nitro Middleware, I try to customize it, I just have to rename it.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by danielroe** [comment, marked as answer]
The Nitro middleware directory is `~/server/middleware`.

Can you explain more about what you want to configure?

**Reply by icai** [reply]
@danielroe if All server I use `fromNodeMiddleware` to import,  The Nitro middleware directory could be not appropriate.

**Reply by icai** [reply]
<img width="834" alt="image" src="https://github.com/nuxt/nuxt/assets/1061012/07ba3fa6-1228-4568-8119-29dd0abe91f0">
@danielroe try to see the project directory architecture.
